build/guide/multi-platform.md

```diff
@@ -0,0 +1,263 @@
+pipeline runs per platform.
+
+![Build pipelines using cross-compilation](./images/cross-compiled-builds.png){:.invertible}
+
+### Update the Dockerfile
+
+To build the app using the cross-compilation technique, update the Dockerfile as
+follows:
+
+- Add `--platform=$BUILDPLATFORM` to the `FROM` instruction for the initial
+  `base` stage, pinning the platform of the `golang` image to match the
+  architecture of the host machine.
+- Add `ARG` instructions for the Go compilation stages, making the `TARGETOS`
+  and `TARGETARCH` build arguments available to the commands in this stage.
+- Set the `GOOS` and `GOARCH` environment variables to the values of `TARGETOS`
+  and `TARGETARCH`. The Go compiler uses these variables to do
+  cross-compilation.
+
+```diff
+  # syntax=docker/dockerfile:1
+- FROM golang:1.20.3-alpine AS base
++ FROM --platform=$BUILDPLATFORM golang:1.20.3-alpine AS base
+  ENV CGO_ENABLED=0
+  WORKDIR /src
+  RUN --mount=type=cache,target=/go/pkg/mod \
+      --mount=type=bind,source=go.mod,target=go.mod \
+      --mount=type=bind,source=go.sum,target=go.sum \
+      go mod download -x
+
+  FROM base AS build-client
++ ARG TARGETOS
++ ARG TARGETARCH
+  RUN --mount=type=cache,target=/go/pkg/mod \
+      --mount=type=bind,target=. \
+-     go build -o /bin/client ./cmd/client
++     GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -o /bin/client ./cmd/client
+
+  FROM base AS build-server
++ ARG TARGETOS
++ ARG TARGETARCH
+  RUN --mount=type=cache,target=/go/pkg/mod \
+      --mount=type=bind,target=. \
+-     go build -o /bin/server ./cmd/server
++     GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -o /bin/server ./cmd/server
+
+  FROM scratch AS client
+  COPY --from=build-client /bin/client /bin/
+  ENTRYPOINT [ "/bin/client" ]
+
+  FROM scratch AS server
+  COPY --from=build-server /bin/server /bin/
+  ENTRYPOINT [ "/bin/server" ]
+
+  FROM scratch AS binaries
+  COPY --from=build-client /bin/client /
+  COPY --from=build-server /bin/server /
+```
+
+The only thing left to do now is to run the actual build. To run a
+multi-platform build, set the `--platform` option, and specify a CSV string of
+the OS and architectures that you want to build for. The following command
+illustrates how to build, and export, binaries for Mac (ARM64), Windows, and
+Linux:
+
+```console
+$ docker buildx build \
+  --target=binaries \
+  --output=bin \
+  --builder=container \
+  --platform=darwin/arm64,windows/amd64,linux/amd64 .
+```
+
+When the build finishes, you’ll find client and server binaries for all of the
+selected platforms in the `bin` directory:
+
+```diff
+bin
+├── darwin_arm64
+│   ├── client
+│   └── server
+├── linux_amd64
+│   ├── client
+│   └── server
+└── windows_amd64
+    ├── client
+    └── server
+```
+
+## Summary
+
+This section has demonstrated how you can get started with multi-platform builds
+using emulation and cross-compilation.
+
+Related information:
```

Comment:
Maybe an explicit callout to https://github.com/tonistiigi/xx would be good here.

Reply:
Ah right I should have read all the comments :sweat_smile: https://github.com/docker/docs/pull/17051#discussion_r1166581018